pkg/metrics/metrics.go

```diff
@@ -0,0 +1,141 @@
+package metrics
+
+import (
+	"net/http"
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/klog/v2"
+)
+
+var (
+	r    *metricRecorder // singleton instance of metricRecorder
+	once sync.Once
+)
+
+type metricRecorder struct {
+	registry metrics.KubeRegistry
+	metrics  map[string]interface{}
+}
+
+// Recorder returns the singleton instance of metricRecorder.
+// nil is returned if the recorder is not initialized.
+func Recorder() *metricRecorder {
+	return r
+}
+
+// NewRecorder initializes a new metricRecorder instance if it hasn't been initialized.
+func NewRecorder() *metricRecorder {
```

Comment:
Thoughts on renaming this to something like `InitializeRecorder`? As-is, I'm not sure the name conveys that not only are you creating a new instance of the recorder object, but that it's going to be set as the singleton.